### PR TITLE
lgtm: replace the query used for looking for fgets with a more general query

### PR DESCRIPTION
**Pull request by evverx**
to make it easier to comlain about `strtok` :-)

Inspired by https://github.com/systemd/systemd/pull/11963, which, in turn, was prompted by https://github.com/systemd/systemd/pull/11555.

### CODE REVIEW

**Comment by evverx** [comment]
Judging by https://lgtm.com/query/4363101629453643551/, it seems to be working.

**Comment by poettering** [comment]
Excellent! Thanks!

File: .lgtm/cpp-queries/PotentiallyDangerousFunction.ql [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+ * @description Certain standard library functions are dangerous to call.
+ * @kind problem
+ * @problem.severity error
+ * @precision high
+ * @id cpp/potentially-dangerous-function
+ * @tags reliability
+ *       security
+ *
+ * Borrowed from
+ * https://github.com/Semmle/ql/blob/master/cpp/ql/src/Security/CWE/CWE-676/PotentiallyDangerousFunction.ql
```
Comment (evverx) [inline .lgtm/cpp-queries/PotentiallyDangerousFunction.ql:12]: The query pack has just been updated and everything seems to be more or less fine except that "borrowed", "from" and the url are considered tags: https://lgtm.com/rules/1507954176130/. @samlanning is this expected and I should move that comment somewhere else?

It's unrelated, but I've just run into https://lgtm.com/blog/introducing_dataflow_path_exploration. I'm wondering if it is the same as "taint source modeling" you were talking about in https://github.com/systemd/systemd/pull/11754#issuecomment-465707129.

Comment (s0) [reply .lgtm/cpp-queries/PotentiallyDangerousFunction.ql:12]: @evverx regarding the tags, yes this is expected. For comments like that, i would suggest just using 'normal' block comments outside of the `QLDoc`, underneath it. i.e. just using `/* */`

Regarding the taint source modelling, the resulting query would probably be a taint-tracking or data-flow query, so yes. The query could be expressed as a `path-problem`, and would produce paths that can be viewable in LGTM's UI.

**Pull request merged**